### PR TITLE
Frontend refactor

### PR DESCRIPTION
**Pull request by karlek**

### CODE REVIEW

File: cmd/pippi/frontend/src/app/disassembly/disassembly.component.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export class DisassemblyComponent {
     if (!data.currentValue) {
       return;
     }
-    this._data = JSON.stringify(data.currentValue);
+    this._data = JSON.stringify(data.currentValue, null, 2);
```
Comment (mewmew) [inline cmd/pippi/frontend/src/app/disassembly/disassembly.component.ts:19]: What do the extra arguments to stringify do? Also, is there a good `jsdoc` tool (and `ngdoc`) to check docs locally?

File: cmd/pippi/frontend/src/app/selection.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { TestBed } from '@angular/core/testing';
+
+import { SelectionService } from './selection.service';
+
+describe('SelectionService', () => {
+  beforeEach(() => TestBed.configureTestingModule({}));
+
+  it('should be created', () => {
+    const service: SelectionService = TestBed.get(SelectionService);
+    expect(service).toBeTruthy();
```
Comment (mewmew) [inline cmd/pippi/frontend/src/app/selection.service.spec.ts:10]: `toBeTruthy` haha :) gotta love assert testing JS style

File: cmd/pippi/frontend/src/app/strings/strings.component.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<ace-editor
```
Comment (mewmew) [inline cmd/pippi/frontend/src/app/strings/strings.component.html:1]: Cool, this is a useful HTML tag/component. Was it part of ace originally, or did you hook it up?

**Review by mewmew — APPROVED** [review]
LGTM, and even with test cases o.O

**Comment by mewmew** [comment]
Is this PR run on Circle? o.O can't find the output.

File: cmd/pippi/frontend/src/app/disassembly/disassembly.component.ts (hunk 1 above)
Comment (karlek) [reply cmd/pippi/frontend/src/app/disassembly/disassembly.component.ts:19]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#Parameters

MDN seems to be available offline, but can't find anything about angular.
https://kapeli.com/mdn_offline

Comment (karlek) [reply cmd/pippi/frontend/src/app/disassembly/disassembly.component.ts:19]: (I haven't tried that last link btw)

There are no well-known offline docs tools, afaik

**Pull request merged**